src/OrchardCore.Modules/OrchardCore.Contents/Controllers/ApiController.cs

```diff
@@ -73,7 +74,13 @@ public async Task<IActionResult> Post(ContentItem newContentItem, bool draft = f
                     return Unauthorized();
                 }
 
-                await _contentManager.CreateAsync(newContentItem, VersionOptions.DraftRequired);
+                // Call ActivatedAsync and InitializedAsync
+                var newContentItem = await _contentManager.NewAsync(model.ContentType);
+                newContentItem.Apply(model);
+
+                // Call UpdatedAsync and UpdatingAsync and CreatedAsync and CreatingAsync
+                // In the same order than we do in the AdminController of OrchardCore.Contents.
+                await _contentManager.UpdateAndCreateAsync(model, draft ? VersionOptions.DraftRequired : VersionOptions.Published);
```

Comment:
If the item already exists, below we still need to publish it if draft is false.

Reply:
Also in my tests in the previous PR (create handlers -> update handlers) if the TitlePart has a liquid pattern and the AutoroutePart has also a liquid pattern that uses DisplayText eg. {ContentItem.DisplayText | slugify } the autoroutepart.part will not get filled. Will work if DisplayText is explicitly set like in the example workflow script i provided.

Reply:
Here we are in a If(contentItem == null)

Reply:
@MichaelPetrinolis Yes we call the Update events so it calls the AutoroutePartHandler which will set the AutoroutePart.Path with the Liquid pattern. So no need to provide an AutoroutePart.Path in the JSON. Though we should probably also test if someone provides it.

Reply:
Ok I tested by providing an AutoroutePart JSON content and it works too. So it will take precedence over the DisplayText and it will also replace the Liquid Pattern. Makes sense to me that if I specify an AutoroutePart that it should take it instead of generating it from the configured liquid pattern. Which means that if we want to have that same pattern we can set a dynamic value from that script. Could be that if the content item is created from a workflow then the autoroute pattern could be set differently. I'm leaving it like that.